runtime/compiler/z/codegen/J9TreeEvaluator.hpp

```diff
@@ -73,6 +73,46 @@ class OMR_EXTENSIBLE TreeEvaluator: public J9::TreeEvaluator
    static TR::Register *conditionalHelperEvaluator(TR::Node *node, TR::CodeGenerator *cg);
    static float interpreterProfilingInstanceOfOrCheckCastTopProb(TR::CodeGenerator * cg, TR::Node * node);
 
+   // Software Concurrent Scavenge evaluators
+   static TR::Register *ardbarEvaluator(TR::Node *node, TR::CodeGenerator *cg);
+   /**
+    * \brief
+    * In concurrent scavenge (CS) mode, this is used to evaluate ardbari nodes and inlined-compare-and-swap to
+    * generate a guarded load software sequence that is functionally equivalent to LGG/LLGFSG instructions on z.
+    *
+    * \param node
+    * \param cg
+    * \param resultReg
+    * \param memRef
+    * \param isCompressedRef
+    * \param hasReturnReg
+    * \param dynLitPooLoad
```

Comment:
There are some undefined parameters here. They also need descriptions. The `\return` needs to be rewritten once `hasReturnReg` gets removed.